// sass-lint-auto-fix/dist/typings/abstract-syntax-tree.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var NodeType;
(function (NodeType) {
    NodeType["arguments"] = "arguments";
    NodeType["atkeyword"] = "atkeyword";
    NodeType["atrule"] = "atrule";
    NodeType["attributeFlags"] = "attributeFlags";
    NodeType["attributeMatch"] = "attributeMatch";
    NodeType["attributeName"] = "attributeName";
    NodeType["attributeSelector"] = "attributeSelector";
    NodeType["attributeValue"] = "attributeValue";
    NodeType["block"] = "block";
    NodeType["brackets"] = "brackets";
    NodeType["class"] = "class";
    NodeType["color"] = "color";
    NodeType["combinator"] = "combinator";
    NodeType["condition"] = "condition";
    NodeType["conditionalStatement"] = "conditionalStatement";
    NodeType["declaration"] = "declaration";
    NodeType["declarationDelimiter"] = "declarationDelimiter";
    NodeType["default"] = "default";
    NodeType["delimiter"] = "delimiter";
    NodeType["dimension"] = "dimension";
    NodeType["escapedString"] = "escapedString";
    NodeType["extend"] = "extend";
    NodeType["expression"] = "expression";
    NodeType["function"] = "function";
    NodeType["global"] = "global";
    NodeType["id"] = "id";
    NodeType["ident"] = "ident";
    NodeType["important"] = "important";
    NodeType["include"] = "include";
    NodeType["interpolatedVariable"] = "interpolatedVariable";
    NodeType["interpolation"] = "interpolation";
    NodeType["keyframesSelector"] = "keyframesSelector";
    NodeType["loop"] = "loop";
    NodeType["mixin"] = "mixin";
    NodeType["multilineComment"] = "multilineComment";
    NodeType["namePrefix"] = "namePrefix";
    NodeType["namespacePrefix"] = "namespacePrefix";
    NodeType["namespaceSeparator"] = "namespaceSeparator";
    NodeType["number"] = "number";
    NodeType["operator"] = "operator";
    NodeType["parentheses"] = "parentheses";
    NodeType["parentSelector"] = "parentSelector";
    NodeType["parentSelectorExtension"] = "parentSelectorExtension";
    NodeType["percentage"] = "percentage";
    NodeType["placeholder"] = "placeholder";
    NodeType["progid"] = "progid";
    NodeType["property"] = "property";
    NodeType["propertyDelimiter"] = "propertyDelimiter";
    NodeType["pseudoClass"] = "pseudoClass";
    NodeType["pseudoElement"] = "pseudoElement";
    NodeType["raw"] = "raw";
    NodeType["ruleset"] = "ruleset";
    NodeType["space"] = "space";
    NodeType["selector"] = "selector";
    NodeType["singlelineComment"] = "singlelineComment";
    NodeType["string"] = "string";
    NodeType["stylesheet"] = "stylesheet";
    NodeType["typeSelector"] = "typeSelector";
    NodeType["unicodeRange"] = "unicodeRange";
    NodeType["universalSelector"] = "universalSelector";
    NodeType["urange"] = "urange";
    NodeType["uri"] = "uri";
    NodeType["value"] = "value";
    NodeType["variable"] = "variable";
    NodeType["variablesList"] = "variablesList";
})(NodeType = exports.NodeType || (exports.NodeType = {}));
